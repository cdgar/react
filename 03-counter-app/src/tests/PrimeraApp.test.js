import React from 'react';
import { render } from "@testing-library/react";
import FirtsApp from "../PrimeraApp";
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';

describe('PrimeraApp.test.js', () => {
    // test('should show Hola desde index', () => {
    //   const saludo = "Hola desde index";
    //   const {getByText} = render(<FirtsApp saludo={saludo}/>)
    //   expect(getByText(saludo)).toBeInTheDocument();
    // })

    test('should show <FirtsApp/>', () => {
        const saludo = "Hola desde index";
        const wrapper = shallow(<FirtsApp saludo={saludo}/>)

        expect(wrapper).toMatchSnapshot();
    })
    test('should show subtitle - valor por defecto', () => {
        const saludo = "Hola desde index";
        const wrapper = shallow(<FirtsApp saludo={saludo}/>)

        const textSubtitle = wrapper.find('p').text();
        expect(textSubtitle).toBe('subtitle - valor por defecto');
    })

})
