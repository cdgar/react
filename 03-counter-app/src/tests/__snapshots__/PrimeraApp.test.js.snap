// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeraApp.test.js should show <FirtsApp/> 1`] = `
<Fragment>
  <h1>
    Hola desde index
  </h1>
  <p>
    subtitle - valor por defecto
  </p>
</Fragment>
`;
